A6-client-retries.md

```diff
@@ -284,21 +282,24 @@ When an RPC is evicted from the buffer, pending hedged requests should be cancel
+RPC failures can occur in four distinct ways:
 
-1. The RPC never leaves the client.
-2. The RPC reaches the server, but has never been seen by the server application logic.
-3. The RPC is seen by the server application logic, and fails.
+1. The RPC fails at the client-side load balancing step.
+2. Load balancing succeeds but the RPC never leaves the client.
+3. The RPC reaches the server, but has never been seen by the server application logic.
+4. The RPC is seen by the server application logic, and fails.
 
 ![Where RPCs Fail](A6_graphics/WhereRPCsFail.png)
 
 [Link to SVG file](A6_graphics/WhereRPCsFail.svg)
 
-The last case is handled by the configurable retry policy that is the main focus of this document. The first two cases are retried automatically by the gRPC client library, **regardless** of the retry configuration set by the service owner. We are able to do this because these request have not made it to the server application logic, and thus are always safe to retry.
+The last case is handled by the configurable retry policy that is the main focus of this document. The second and third cases are retried automatically by the gRPC client library, **regardless** of the retry configuration set by the service owner. We are able to do this because these request have not made it to the server application logic, and thus are always safe to retry.
+
+In the first case, in which the RPC fails at the load balancing step, the appropriate action depends on what pick result type caused that failure. If the pick result type was `DROP`, the RPC will fail immediately, without any retries. If the pick result type was `TRANSIENT_FAILURE`, the RPC will be handled by the configured retry policy.
 
-In the first case, in which the RPC never leaves the client, the client library can transparently retry until a success occurs, or the RPC's deadline passes.
+In the second case, in which load balancing succeeds but the RPC never leaves the client, the client library can transparently retry until a success occurs, or the RPC's deadline passes.
 
-If the RPC reaches the gRPC server library, but has never been seen by the server application logic (the second case), the client library will immediately retry it once. If this fails, then the RPC will be handled by the configured retry policy. This extra caution is needed because this case involves extra load on the wire.
+If the RPC reaches the gRPC server library, but has never been seen by the server application logic (the third case), the client library will immediately retry it once. If this fails, then the RPC will be handled by the configured retry policy. This extra caution is needed because this case involves extra load on the wire. The client knows it is in this case if the stream ends with an `RST_STREAM` frame with the error code `REFUSED_STREAM` or if the HTTP/2 connection closes with a `GOAWAY` frame with the last stream identifier less than the stream's ID.
```

Comment:
The new sentence at the end of this paragraph is specific to the HTTP/2 transport, but there may be other transports, so I think we need to explain this in that context.  I suggest starting the sentence with "In the gRPC HTTP/2 transport, this occurs when".

Reply:
Done, with a slightly different wording.